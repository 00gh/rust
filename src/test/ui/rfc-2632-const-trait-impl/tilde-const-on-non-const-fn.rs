#![feature(const_trait_impl)]
fn a<T: ~const From<u8>>() {}
//~^ ERROR: `~const` is not allowed

struct S;

impl S {
    fn b<T: ~const From<u8>>() {}
    //~^ ERROR: `~const` is not allowed
}

fn main() {}
