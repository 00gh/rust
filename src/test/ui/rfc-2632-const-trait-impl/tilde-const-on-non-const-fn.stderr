error: `~const` is not allowed here
  --> $DIR/tilde-const-on-non-const-fn.rs:2:9
   |
LL | fn a<T: ~const From<u8>>() {}
   |         ^^^^^^^^^^^^^^^
   |
   = note: only allowed on bounds on traits' associated types, const fns, const impls and its associated functions

error: `~const` is not allowed here
  --> $DIR/tilde-const-on-non-const-fn.rs:8:13
   |
LL |     fn b<T: ~const From<u8>>() {}
   |             ^^^^^^^^^^^^^^^
   |
   = note: only allowed on bounds on traits' associated types, const fns, const impls and its associated functions

error: aborting due to 2 previous errors

